### PR TITLE
Sync peep spawn across multiplayer

### PR DESCRIPTION
**Pull request by wolfreak99**
There was also some minor adjustments here and there throughout, but I mostly followed the layout of placing a park entrance. I'm exhausted and really tired, I'll update this message later with more information.

### CODE REVIEW

**Comment by Gymnasiast** [comment]
When you're fresh again, could you rebase and try to fix the CIs?

File: distribution/changelog.txt [hunk 1]
```diff
@@ -18,6 +18,7 @@
 - Fix: [#5981] Ride list doesn't update after using quick demolish.
 - Fix: [#5984] Allow socket binding to same port after crash
 - Fix: [#5998] Staff not getting paid / no loan interest.
+- Fix: [#5901] Placing peep spawn not synced across multiplayer
```
Comment (Gymnasiast) [inline distribution/changelog.txt]: Needs to be higher up the list. Changelog is order by category, then issue number.

**Comment by Gymnasiast** [comment]
Code looks reasonable. Have you tested it with multiplayer?

File: src/openrct2/windows/map.c [hunk 2]
```diff
@@ -1349,41 +1347,32 @@ static void window_map_place_park_entrance_tool_down(sint32 x, sint32 y)
...
-    surfaceMapElement = map_get_surface_element_at(mapX >> 5, mapY >> 5);
-    if (surfaceMapElement == NULL) {
-        return;
-    }
-    if (surfaceMapElement->properties.surface.ownership & 0xF0) {
-        return;
-    }
-
-    mapX = mapX + 16 + (word_981D6C[direction].x * 15);
-    mapY = mapY + 16 + (word_981D6C[direction].y * 15);
```
Comment (wolfreak99) [inline src/openrct2/windows/map.c]: While working on this pull request I began to have some issues with getting map_get_path_element_at to work correctly in the `map_place_peep_spawn` function, and this portion got removed. 

I'm not sure of the original intention of `word_981D6C` or if that portion was even necessary, _and come to think of it I never did test this in various map rotations/directions_, as after the troubles I was just happy with getting map_get_path_element_at to work properly.

In a moment, I'm going to test add
```
sint32 mapXOld = mapX + 16 + (word_981D6C[direction].x * 15);		
sint32 mapYOld = mapY + 16 + (word_981D6C[direction].y * 15);
assert(mapX == mapXOld && mapY == mapYOld);
```
and see if the values are different in various map rotations, directions, and sides of the world. Should there be no assertion errors, should I push the above code in with a comment?

Comment (Gymnasiast) [inline src/openrct2/windows/map.c]: Er, I don't know.
@IntelOrca ?

Comment (IntelOrca) [inline src/openrct2/windows/map.c]: This is setting the map position to be one unit in front of the edge (of `direction`). Leave it be.

File: src/openrct2/world/park.c [hunk 3]
```diff
@@ -111,8 +111,6 @@ void park_init()
     reset_researched_scenery_items();
 
     gParkEntranceFee = MONEY(10, 00);
-    gPeepSpawns[0].x = PEEP_SPAWN_UNDEFINED;
-    gPeepSpawns[1].x = PEEP_SPAWN_UNDEFINED;
```
Comment (IntelOrca) [inline src/openrct2/world/park.c]: Why have remove these?

Comment (wolfreak99) [inline src/openrct2/world/park.c]: The only time `park_init()` is called is during `game_init_all(sint32 mapsize)`, which also calls `reset_park_entry()`. `reset_park_entry()` sets all peep spawns to null, so there is no sense in having the first 2 peep spawns reset, only to later have all peep spawns reset in the same function.

I meant to make a seperate branch for this but I have not learned yet how to make a branch of only a few lines of code from a file. (if it's even possible).

File: src/openrct2/windows/map.c (hunk 2 above)
Comment (wolfreak99) [inline src/openrct2/windows/map.c]: I'm not sure how this is happening, but 
```
if (mapX != MAP_LOCATION_NULL) {
sint32 mapXOld, mapYOld;
mapXOld = mapX + 16 + (word_981D6C[direction].x * 15);
mapYOld = mapY + 16 + (word_981D6C[direction].y * 15);
assert(mapX == mapXOld && mapY == mapYOld);
mapZ = mapElement->base_height / 2;
```
Is not triggering any assertion errors. At the same time, having.
```
if (mapX != MAP_LOCATION_NULL) {
mapX = mapX + 16 + (word_981D6C[direction].x * 15);
mapY = mapY + 16 + (word_981D6C[direction].y * 15);
mapZ = mapElement->base_height / 2;
```
Causes the placement to not work. I'm not sure how this is happening, it doesn't make sense to me. Can I push the version with the assert comparison and you guys tell me if you can trigger the assert?

Comment (wolfreak99) [inline src/openrct2/windows/map.c]: @IntelOrca @Gymnasiast Could one of you guys give me an update if you guys were able to get that assert that i mentioned above? i'm not sure if you got the reply for this or not.

Also @Gymnasiast , in response to "have you tested this in multiplayer" Yes I did, I had to test it with localhost and advertising disabled, and tested it from the same computer, but it did work as behaved before switched it back to the old formula (which is what's confusing me above).

**Comment by Gymnasiast** [comment]
Could you rebase this?

**Comment by wolfreak99** [comment]
@Gymnasiast 
Will do once I get home from my meeting in like 3 or so hours. Would you be able to see if you can get that assert to trigger in the meantime with the code as is?

**Comment by wolfreak99** [comment]
@Gymnasiast @IntelOrca I just realized that the whole `mapX + 16 + (word_981D6C[direction].x * 15)` was actually what makes peeps show up in the _middle_ of the tile, rather than the corner of the tile. I also cleaned up the commit history a bit more. This now works fabulously.

**Comment by duncanspumpkin** [comment]
This requires rework to be converted to a Game Action. This shouldn't be too hard to do as it doesn't get used in tracks. Look at the game action to place park entrances for tips on how to do it.

**Comment by wolfreak99** [comment]
Would I be able to squash the existing commits together first?

**Comment by duncanspumpkin** [comment]
Sure go ahead.

**Comment by wolfreak99** [comment]
Hey guys, I've been absent for a few months because I'm out on my own now and have been working alot of hours, but I'm going to attempt to dive in and see if I can resurrect this. I will likely make a new branch since the code has likely been updated alot since I last was active.

**Comment by duncanspumpkin** [comment]
For the new version can you try make it a game action. Don't really want any more game commands since they are the old format

**Comment by wolfreak99** [comment]
@duncanspumpkin I'm about to post a PR of the newer branch to see if it compiles okay (my side is giving me a bunch of weird external symbol issues with the dll, but I don't have much time right now to investigate all of that), But I am assuming you are asking to not include all of the stuff prefixed with "game_command" etc, right? A possible issue I could see needing help would be with the network log and game action permissions group portion, other than that I could likely give it a shot. Once it compiles to make sure I did it right

**Comment by Broxzier** [comment]
Superseded by #7047.

**Pull request closed without merging**